src/stream/stream/timeout.rs

```diff
@@ -0,0 +1,59 @@
+use std::error::Error;
+use std::fmt;
+use std::pin::Pin;
+use std::time::Duration;
+
+use futures_timer::Delay;
+
+use crate::future::Future;
+use crate::stream::Stream;
+use crate::task::{Context, Poll};
+
+#[doc(hidden)]
+#[derive(Debug)]
+pub struct TimeoutStream<S> {
```

Comment:
We should rename this to `Timeout` so it can be exported + enumerated from the list of streams, similar to all the other streams.